File: topic_test/scripts/my_test_subscriber.py
#!/usr/bin/python
#-*- coding: utf-8 -*-

import rospy
from std_msgs.msg import Int64

def msgCallback(msg):
    rospy.loginfo("msg : %d", msg.data)

def my_listener():
    rospy.init_node("my_test_subscriber")
    rospy.Subscriber("my_count", Int64, msgCallback)

    rospy.spin()

if __name__ == "__main__":
    my_listener()

File: topic_test/src/my_test_subscriber.cpp
#include "ros/ros.h"
#include "std_msgs/Int64.h"

void msgCallback(const std_msgs::Int64::ConstPtr& msg)
{
    ROS_INFO("%d", msg->data);
}

int main(int argc, char** argv)
{
    ros::init(argc, argv, "my_test_subscriber");
    ros::NodeHandle nh;

    ros::Subscriber sub = nh.subscribe("test_topic", 100, msgCallback);

    ros::spin();

    return 0;
}